pkg/certificate/identity.go

```diff
@@ -287,9 +288,14 @@ func (h *identityHandler) GetX509RoleCert() (rolecerts [](*RoleCertificate), rol
 		if err != nil {
 			return nil, nil, fmt.Errorf("Failed to prepare csr, failed to parse certificate for PostRoleCertificateRequest, Subject CommonName[%s], err: %v", csrOption.Subject.CommonName, err)
 		}
+
+		var roleCertExpiry int64 // 0 implies using server default expiry time
+		if h.idCfg.ServiceCert.CopperArgos.Use {
+			roleCertExpiry = int64(x509LeafCert.NotAfter.Sub(time.Now()).Minutes()) + int64(config.DEFAULT_ROLE_CERT_EXPIRY_TIME_BUFFER_MINUTES) // Extract NotAfter from the instance certificate
+		}
 		roleRequest := &zts.RoleCertificateRequest{
 			Csr:        string(roleCsrPEM),
-			ExpiryTime: int64(x509LeafCert.NotAfter.Sub(time.Now()).Minutes()) + int64(config.DEFAULT_ROLE_CERT_EXPIRY_TIME_BUFFER_MINUTES), // Extract NotAfter from the instance certificate
+			ExpiryTime: roleCertExpiry,
```

Comment:
(Optional) Consider using anonymous function to make it more readable
```
ExpiryTime: func() {
  return ...
}(),
```

Reply:
fixed in [fa2aae3](https://github.com/AthenZ/k8s-athenz-sia/pull/178/commits/fa2aae34bae2492317f3fa61d96315f58ff745f7)